src/utils/transform.ts

```diff
@@ -197,3 +216,293 @@ function valueToTag(key: string, value: string | number | undefined, type: strin
     value,
   };
 }
+
+/**
+ * Get data frame to be shown in NodeGraph in Grafana with all the required stats. As we process both service map
+ * requests and trace graph requests here you have to set whether to show traces per minute stat or request count as
+ * trace graph cannot compute traces pre minute.
+ */
+export function parseGraphResponse(response: DataFrame, query?: XrayQuery, options?: { showRequestCounts: boolean }) {
+  // Again assuming this will be single field with single value which will be the trace data blob
+  const services: XrayService[] = response.fields[0].values.toArray().map(serviceJson => {
+    return JSON.parse(serviceJson);
+  });
+
+  const showRequestCounts = options?.showRequestCounts ?? false;
+
+  const idField = {
+    name: 'id',
+    type: FieldType.string,
+    values: new ArrayVector(),
+  };
+  const titleField = {
+    name: 'name',
+    type: FieldType.string,
+    values: new ArrayVector(),
+    // TODO: use constants from grafana/ui
+    labels: { NodeGraphValueType: 'title' },
+    config: { displayName: 'Name' },
+  };
+
+  const typeField = {
+    name: 'type',
+    type: FieldType.string,
+    values: new ArrayVector(),
+    labels: { NodeGraphValueType: 'subTitle' },
+    config: { displayName: 'Type' },
+  };
+
+  const mainStatField = {
+    name: 'average_response_time',
+    type: FieldType.number,
+    values: new ArrayVector(),
+    config: { unit: 'ms/t' },
+    labels: { NodeGraphValueType: 'mainStat' },
+  };
+
+  const secondaryStatField = showRequestCounts
+    ? {
+        name: 'requests_count',
+        type: FieldType.string,
+        values: new ArrayVector(),
+        labels: { NodeGraphValueType: 'secondaryStat' },
+      }
+    : {
+        name: 'transactions_per_minute',
+        type: FieldType.number,
+        values: new ArrayVector(),
+        config: { unit: 't/min' },
+        labels: { NodeGraphValueType: 'secondaryStat' },
+      };
+
+  const successField = {
+    name: 'success',
+    type: FieldType.number,
+    values: new ArrayVector(),
+    labels: { NodeGraphValueType: 'arc' },
+    config: { color: { fixedColor: 'rgb(80, 171, 113)' } },
```

Comment:
Is there a way to use named color instead: https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/utils/namedColorsPalette.ts 

they adapt slightly a bit for light & dark theme

Reply:
Yes I can map it, these are taken directly from x-ray console.